operator/v1/types.go

```diff
@@ -0,0 +1,230 @@
+type LoggingConfig struct {
+	Type string `json:"type"`
+
+	Glog     *GlogConfig     `json:"glog,omitempty"`
+	CapnsLog *CapnsLogConfig `json:"capnsLog,omitempty"`
+	Java     *JavaLog        `json:"java,omitempty"`
+}
+
+// GlogConfig holds information about configuring logging
+type GlogConfig struct {
+	// level is passed to glog.
+	Level int64 `json:"level"`
+
+	// vmodule is passed to glog.
+	Vmodule string `json:"vmodule"`
+}
+
+type CapnsLogConfig struct {
+	// level is passed to capnslog: critical, error, warning, notice, info, debug, trace
+	Level string `json:"level"`
+
+	// There is some kind of repo/package level thing for this
+}
+
+type JavaLog struct {
+	// level is passed to jsr47: fatal, error, warning, info, fine, finer, finest
+	Level string `json:"level"`
+
+	// There is some kind of repo/package level thing for this.  might end up hierarchical
+}
+
+type OperatorStatus struct {
+	// conditions is a list of conditions and their status
+	Conditions []OperatorCondition `json:"conditions,omitempty"`
+
+	// version is the level this availability applies to
+	Version string `json:"version"`
+
+	// readyReplicas indicates how many replicas are ready and at the desired state
+	ReadyReplicas int32 `json:"readyReplicas"`
+
+	// generations are used to determine when an item needs to be reconciled or has changed in a way that needs a reaction.
+	Generations []GenerationStatus `json:"generations"`
+}
+
+// GenerationStatus keeps track of the generation for a given resource so that decisions about forced updated can be made.
+type GenerationStatus struct {
+	// group is the group of the thing you're tracking
+	Group string `json:"group"`
+	// resource is the resource type of the thing you're tracking
+	Resource string `json:"resource"`
+	// namespace is where the thing you're tracking is
+	Namespace string `json:"namespace"`
+	// name is the name of the thing you're tracking
+	Name string `json:"name"`
+	// lastGeneration is the last generation of the workload controller involved
+	LastGeneration int64 `json:"lastGeneration"`
+	// hash is an optional field set for resources without generation that are content sensitive like secrets and configmaps
+	Hash string `json:"hash"`
+}
+
+var (
+	// Available indicates that the operand is present and accessible in the cluster
+	OperatorStatusTypeAvailable = "Available"
+	// Progressing indicates that the operator is trying to transition the operand to a different state
+	OperatorStatusTypeProgressing = "Progressing"
+	// Failing indicates that the operator (not the operand) is unable to fulfill the user intent
+	OperatorStatusTypeFailing = "Failing"
+	// PrereqsSatisfied indicates that the things this operator depends on are present and at levels compatible with the
+	// current and desired states.
+	OperatorStatusTypePrereqsSatisfied = "PrereqsSatisfied"
+	// Upgradeable indicates that the operator configuration itself (not prereqs) can be auto-upgraded by the CVO
```

Comment:
PrereqsSatisfied could be False, but Upgradeable could be True, but only if i've set ManagementState to Force?

PrereqsSatisfied could be True, but Upgradeable can be False if I've done something to make my configuration invalid?